lib/plugins/aws/lib/monitorStack.js

```diff
@@ -25,127 +24,131 @@ module.exports = {
 
     this.serverless.cli.log(`Checking Stack ${action} progress...`);
 
-    return new BbPromise((resolve, reject) => {
-      async.whilst(
-        () => validStatuses.indexOf(stackStatus) === -1,
-        callback => {
-          const cfStackStatusCheckFrequency =
-            options.frequency || process.env.SLS_AWS_MONITORING_FREQUENCY || 5000;
+    return new Promise((resolve, reject) => {
+      const recursivePromiseFn = () => {
+        if (validStatuses.indexOf(stackStatus) === -1) {
+          return (() => {
+            return new Promise(callback => {
```

Comment:
Instead of `new Promise` `setTimeout` wrapper, let's use `BbPromise.delay` instead.

I know that `BbPromise.delay` is not standard, we will have to anyway replace it with some common _wait_ utility. Still at this point (as we don't have yet common _wait_ utility coined) I think it's fine to just use bluebird, and leave it for other task.